src/inmanta/data/__init__.py

```diff
@@ -331,26 +333,15 @@ def coalesce_to_min(self, value_reference: str) -> str:
 
         return value_reference
 
+    def with_prefix(self, table_prefix: Optional[str]) -> "ColumnType":
+        return ColumnType(self.base_type, self.nullable, table_prefix)
 
-class TablePrefixWrapper(ColumnType):
-    def __init__(self, table_name: str, child: ColumnType) -> None:
-        self.table_name = table_name
-        self.child = child
 
-    @property
-    def nullable(self) -> bool:
-        return self.child.nullable
-
-    def get_value(self, value: object) -> Optional[PRIMITIVE_SQL_TYPES]:
-        return self.child.get_value(value)
-
-    def get_accessor(self, column_name: str, table_prefix: Optional[str] = None) -> str:
-        if not table_prefix:
-            table_prefix = self.table_name
-        return self.child.get_accessor(column_name, table_prefix)
-
-    def coalesce_to_min(self, value_reference: str) -> str:
-        return self.child.coalesce_to_min(value_reference)
+def TablePrefixWrapper(table_name: Optional[str], child: ColumnType) -> ColumnType:
```

Comment:
I don't like that the column name is not in here, but the prefix is, would require full refactor.